examples/lib/stories/collision_detection/raycast_example.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/geometry.dart';
+import 'package:flame/input.dart';
+import 'package:flame/palette.dart';
+import 'package:flutter/material.dart';
+
+class RaycastExample extends FlameGame
+    with HasCollisionDetection, TapDetector, MouseMovementDetector {
+  static const description = '''
+In this example the raycast functionality is showcased, if you move the mouse
+around the canvas the rays will be cast from its location. You can also tap to
+create a permanent source of rays that wont move with with mouse.
+  ''';
+
+  Ray2? ray;
+  Ray2? reflection;
+  Vector2? origin;
+  Vector2? tapOrigin;
+  bool isOriginCasted = false;
+  bool isTapOriginCasted = false;
+  Paint paint = Paint()..color = Colors.amber.withOpacity(0.2);
+  Paint tapPaint = Paint()..color = Colors.blue.withOpacity(0.2);
+
+  static const numberOfRays = 2000;
```

Comment:
That was the point, that it should look like a light source since that is a common usecase for it.
But you do have a point, I'll rename this example and create a new one with fewer rays.

Reply:
Oh is it? I haven't realized.
When we write documentation for this feature, it would be useful to discuss potential ways to apply it.

In this case, in order to create a light-source effect, would you draw these rays on a separate layer, and then apply via a decorator as a transparency mask?

Reply:
> In this case, in order to create a light-source effect, would you draw these rays on a separate layer, and then apply via a decorator as a transparency mask?

I haven't actually thought about it, but that sounds like a good idea.